shm/Player.cpp

```diff
@@ -22,8 +21,8 @@ size_t Player::countAvailableSpace() {
     auto occupiedSpace = std::accumulate(begin(ship_->getAllCargos()),
                                          end(ship_->getAllCargos()),
                                          0,
-                                         [](const Cargo& cargoFirst, const Cargo& cargoSecond) {
-                                             return cargoFirst.getAmount() + cargoSecond.getAmount();
+                                         [](int accumulated, const auto cargoSecond) {
```

Comment:
Why `auto` and not `auto&` ? Would it be better to use reference here or not?

Reply:
Changed. Tbh my priority was delegate, fixing this was an afterthought because i couldn't test delegate without fixing this function